File: client/src/app/user-auth/assemble.user..ts
export class AssembleUser {

  private _id: string;
  private _email: string;
  private _authenticated: boolean;

  constructor(email: string, id: string, authenticated: boolean) {
    this._email = email;
    this._id = id;
    this._authenticated = false;
  }


  get email(): string {
    return this._email;
  }

  get authenticated(): boolean {
    return this._authenticated;
  }


  get id(): string {
    return this._id;
  }
}

File: assemble-web/user-api/src/main/scala/org/stacktrace/yo/user/model/AssembleUser.scala
package org.stacktrace.yo.user.model

import play.api.libs.json.{Json, OFormat}

/**
  * The user object.
  *
  * @param id       The unique ID of the user.
  * @param email Maybe the email of the authenticated user.
  * @param username    Maybe the name of the authenticated provider.
  */
case class AssembleUser(id: String, email: String, username: Option[String])

object AssembleUser {
  implicit val userFormats: OFormat[AssembleUser] = Json.format[AssembleUser]
}


